collection-scripts/gather_sub

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+DIR_NAME=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+# shellcheck disable=SC1091
+source "${DIR_NAME}/common.sh"
+
+IFS=$'\n'
+
+NS="$1"
+if [ -z "$NS" ]; then
+    echo "No namespace passed, using the default one"
+    NS=openstack
+fi
+
+# Only get resources if the namespace exists
+if ! check_namespace "${NS}"; then
+    exit 0
+fi
+
+
+mkdir -p "${NAMESPACE_PATH}"/"${NS}"
+/usr/bin/oc get packagemanifest -n "$NS" -o yaml >>"${NAMESPACE_PATH}"/"${NS}"/packagemanifests
+
+mkdir -p "${NAMESPACE_PATH}/${NS}/subscriptions/"
+for service in $(oc get subscriptions -n "$NS" -o=custom-columns=NAME:.metadata.name --no-headers)
+do
+    /usr/bin/oc get subscription "$service" -n "$NS" -o yaml >> "${NAMESPACE_PATH}/${NS}/subscriptions/$service"
+done
+
+mkdir -p "${NAMESPACE_PATH}/${NS}/installplans/"
+for service in $(oc get installplan -n "$NS" -o=custom-columns=NAME:.metadata.name --no-headers)
```

Comment:
@fmount 

One thing I want to confirm is whether we need to capture OpenStack must-gather along with OCP must-gather, or if we should only rely on OpenStack-must-gather for all OpenStack-related things.

For example, please check link [1][2]. We have already captured the install plan, subscription, catalog source, CSV (Cluster Service Version), operator condition, and operator group for each namespace in OCP must-gather.

Should we repeat the collection of these items in OpenStack must-gather, or should we rely on OCP must-gather to gather these details?

[1]. https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40096/rehearse-40096-pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1674096584003948544/artifacts/openstack-operator-build-deploy-kuttl/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-0da5ac4ccc0dd27b39d98564c43753501630de95e8ff3b173a8263df4ef651cd/namespaces/openstack-operators/operators.coreos.com/

[2] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40096/rehearse-40096-pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1674096584003948544/artifacts/openstack-operator-build-deploy-kuttl/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-0da5ac4ccc0dd27b39d98564c43753501630de95e8ff3b173a8263df4ef651cd/cluster-scoped-resources/operators.coreos.com/operators/

Reply:
In theory my idea is to ask customers to just run the `must-gather`  only once (with a focus on the OSP related bits), however it makes sense to capture namespaces related to `OLM`, but I guess that's a different conversation. We have two options then:

1. Ask people to run `oc adm must-gather --image=quay.io/<namespace>/must-gather:<tag>` and eventually rely on SOS reports to close the gap on any limitation of the tool (e.g. get nodes related things)
2. Ask people to run both the default OCP must-gather *AND* the OSP one

I'd be more inclined to pursue option 1., but we can talk more about it and figure out what the best strategy is.